src/lib/structures/Argument.ts

```diff
@@ -6,17 +6,93 @@ import type { Awaited } from '../utils/Types';
 import { BaseAliasPiece } from './base/BaseAliasPiece';
 import type { Command } from './Command';
 
+/**
+ * Defines a synchronous result of an [[Argument]], check [[AsyncArgumentResult]] for the asynchronous version.
+ */
 export type ArgumentResult<T> = Awaited<Result<T, UserError>>;
+
+/**
+ * Defines an asynchronous result of an [[Argument]], check [[ArgumentResult]] for the synchronous version.
+ */
 export type AsyncArgumentResult<T> = Promise<Result<T, UserError>>;
 
 export interface IArgument<T> {
+	/**
+	 * The name of the argument, this is used to make the identification of an argument easier.
+	 */
 	readonly name: string;
+
+	/**
+	 * The method which is called when invoking the argument.
+	 * @param argument The argument to parse.
+	 * @param context The context for the method call, contains the message, command, and other options.
+	 */
 	run(argument: string, context: ArgumentContext): ArgumentResult<T>;
 }
 
+/**
+ * The base argument class. This class is abstract and is to be extended by subclasses implementing the methods. In
+ * Sapphire's workflow, arguments are called when using [[Args]]'s methods (usually used inside [[Command]]s by default).
+ *
+ * @example
+ * ```typescript
+ * // TypeScript:
+ * import { Argument, ArgumentResult, PieceContext } from 'sapphire/framework';
+ * import { URL } from 'url';
+ *
+ * // Define a class extending `Argument`, then export it.
+ * // NOTE: You can use `export default` or `export =` too.
+ * export class CoreArgument extends Argument<URL> {
+ *   public constructor(context: PieceContext) {
+ *     super(context, { name: 'hyperlink', aliases: ['url'] });
+ *   }
```

Comment:
Reference `@ApplyOptions` from `import { ApplyOptions } from '@sapphire/decorators'` here

Reply:
Maybe another example? I wanted to show both versions in an 1:1 comparison, alternatively we can document the possible usage of decorators for this purpose.

Reply:
Either of those solutions are fine as well, whichever you prefer